src/libstd/thread/mod.rs

```diff
@@ -1308,13 +1308,17 @@ impl<T> JoinHandle<T> {
         &self.0.thread
     }
 
-    /// Waits for the associated thread to finish.
+    /// Waits for the associated thread to finish. In terms of [atomic memory orderings],
```

Comment:
Could you move the new text to a separate paragraph? The first paragraph shows up as a summary in search results so we like to keep it short. LGTM otherwise.